internal/commands/snap.go

```diff
@@ -7,8 +7,9 @@ type SnapArgs struct {
 
 // SnapOpts holds all flags that are specific to the snap command
 type SnapOpts struct {
-	DisableConsoleConf bool `long:"disable-console-conf" description:"Disable console-conf on the resulting image."`
-	FactoryImage       bool `long:"factory-image" description:"Hint that the image is meant to boot in a device factory."`
+	DisableConsoleConf bool   `long:"disable-console-conf" description:"Disable console-conf on the resulting image."`
+	FactoryImage       bool   `long:"factory-image" description:"Hint that the image is meant to boot in a device factory."`
+	Validation         string `long:"validation" description:"Control whether validations should be ignored or enforced" choice:"ignore" choice:"enforce"`
```

Comment:
I have chosen not to include a default value, so that a blank value can be passed to snapd and snapd can handle the warnings and default values itself.